code/src/components/ThoughtInputBox.js

```diff
@@ -0,0 +1,37 @@
+
+import React, { useState } from 'react';
+import SubmitButton from './SubmitButton';
+
+const ThoughtInputBox = ({ setAllThoughts }) => {
+  const [input, setInput] = useState('');
+
+  const handleInputChange = (event) => {
+    setInput(event.target.value);
+  }
+
+  return (
+    <div className="thoughtInputBox-div">
+      <form>
+        <label
+          className="input-field"
+          htmlFor="input"><h3>What&apos;s making you happy right now?</h3>
+          <textarea
+            id="input"
+            className="input-field"
+            type="text"
+            rows="10"
+            value={input}
+            onChange={handleInputChange}
+            // placesholder decides what the textarea should say before any input is given.
+            placeholder="Add your Happy thoughts here!" />
+          <div className="letter-counter">
+            <p>{140 - input.length} characters left</p>
+          </div>
+        </label>
+      </form>
+      <SubmitButton input={input} setInput={setInput} setAllThoughts={setAllThoughts} />
+    </div>
+  )
+}
+
+export default ThoughtInputBox
```

Comment:
I seriously love your structure, you declare all you need in the respective component and then just insert the component in app. That is so smart to slip transferring props and state, it gives me headache. I will try to do it like you do it for all components. Super smart and clear code!